### PR TITLE
Unify error response format per #317

### PR DESCRIPTION
**Pull request by karcass**
### Explain pull request

From #317 

> Adopt same error responses as for Agency API

### Is this a breaking change

 * No, not breaking

### `Provider` or `agency`

Which API(s) will this pull request impact:

 * `provider`

### Additional context

Nothing immediately comes to mind

### CODE REVIEW

File: provider/README.md [hunk 1]
```diff
@@ -75,6 +75,31 @@ Response bodies must be a `UTF-8` encoded JSON object and must minimally include
...
+* **400:** Bad request.
+* **401:** Unauthorized: Invalid, expired, or insufficient scope of token.
+* **404:** Not Found: Object(s) do not exist.
+* **500:** Internal server error.
+
+#### Error Format
+
+```json
+{
+    "version": "x.y.z",
```
Comment (thekaveman) [inline provider/README.md]: I have a feeling that including `version` in the error response might be more confusing than it is worth. E.g. is that the version that was requested? Or the version that the Provider would have responded with had the request been successful? Or...?

File: provider/README.md [hunk 2]
```diff
@@ -75,6 +75,31 @@ Response bodies must be a `UTF-8` encoded JSON object and must minimally include
 }
 ```
 
+#### Provider HTTP Codes
```
Comment (thekaveman) [inline provider/README.md]: Probably don't need the word *Provider* here. Maybe *HTTP Response Codes*?

Comment (karcass) [reply provider/README.md]: Sure.

File: provider/README.md (hunk 1 above)
Comment (karcass) [reply provider/README.md]: If I take it out, I'll have to modify agency too (which is fine).

**Comment by karcass** [comment]
@thekaveman updated

**Pull request merged**